webapp/graphite/render/functions.py

```diff
@@ -1312,7 +1312,7 @@ def movingWindow(requestContext, seriesList, windowSize, func='average', xFilesF
 
     effectiveXFF = xFilesFactor if xFilesFactor is not None else series.xFilesFactor
 
-    for i in range(windowPoints, len(series)):
+    for i in range(windowPoints + 1, len(series) + 1):
       nonNull = [v for v in series[i - windowPoints:i] if v is not None]
```

Comment:
This is a tricky bit to wrap one's mind around, but I think this is correct.

The highest value `range(windowPoints + 1, len(series) + 1)` returns is `len(series)`, and then just below that's used as the upper bound in the last slice of `series` which ends up with highest index `len(series) - 1`, which is what we expect. 

A minimal demo:

```
>>> series = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h']
>>> window = 3
>>> for i in range(window+1, len(series)+1):
...     print(i, series[i-window: i])
... 
4 ['b', 'c', 'd']
5 ['c', 'd', 'e']
6 ['d', 'e', 'f']
7 ['e', 'f', 'g']
8 ['f', 'g', 'h']
```